### ISSUE TITLE
[bug] RestV1Methods is instantiated missing config argument

### ISSUE THREAD

**Issue by joshvazquez** [conan-io/conan#11268]
https://github.com/conan-io/conan/blob/0eb405bde999d5b7b8e706b206940d2e24aa828e/conans/client/rest/rest_client_common.py#L61-L64

https://github.com/conan-io/conan/blob/0eb405bde999d5b7b8e706b206940d2e24aa828e/conans/client/rest/rest_client.py#L112-L114

The instantiation of RestV1Methods is missing `self._config` as the 6th (excluding self) argument to `__init__` (defined in RestV1Methods's  parent class RestCommonMethods), with `self._verify_ssl` in its place and everything after that shifted over as well. Because of this, the config cannot be used in the `authenticate` pathway.

**Comment by memsharded** [comment]
Hi @joshvazquez 

Thanks for reporting this. Could you please clarify how this can be reproduced from the command line? What errors are you seeing and how you got to this point?

**Comment by joshvazquez** [comment]
@memsharded I am working on a fork with cookie-based authentication to remotes and I encountered this while wanting to access `self._config` in `RestCommonMethods.check_credentials`.

This bug has no impact in the current version because the `RestV1Methods` instantiated by `RestApiClient.authenticate` never accesses `self._config`. The bug only appeared when I modified `RestCommonMethods` to access `self._config`.
